test/core/address_utils/parse_address_test.cc

```diff
@@ -129,7 +129,7 @@ static void test_grpc_parse_ipv6_invalid(const char* uri_text) {
 }
 
 int main(int argc, char** argv) {
-  grpc::testing::TestEnvironment env(argc, argv);
+  grpc::testing::TestEnvironment env(&argc, argv);
```

Comment:
why is changing the signature of TestEnvironment important?  Vast majority of the changes in this PR is the change of this line, and yet this mechanic changes isn't contained in a separate commit, which makes it hard to review the rest of the change (needle in haystack).
Can you include all the refactorings in a separate commit (separate from the actual logic of the change)?

Reply:
> why is changing the signature of TestEnvironment important?

There are situations where command line arguments need to be consumed, not just parsed. In the microbenchmarks, for example, the benchmark framework will complain if an unknown argument is present. Making `argc` a reference argument is a fairly lightweight way to accomplish that.

> Can you include all the refactorings in a separate commit 

It would be harder to justify the refactoring outside this PR. This PR gives it a new concrete usage. I imagine your pushback would be stronger if that PR landed in your review queue as a separate change without its context. If you agree the change would be useful, I can go through the exercise of extracting it to make its review easier.

Reply:
I tend to agree that this seems a little invasive.  Is there an easy way to work around this in the benchmark tests?  If not, this approach is workable, but it would be nice to touch fewer files here.

Reply:
Could we just divorce the TestEnvironment bits from the microbenchmarks?
A quick survey says that they're probably not overly benefiting from this abstraction.

Reply:
> Could we just divorce the TestEnvironment bits from the microbenchmarks?

I think so. However the microbenchmarks were not the only tests sensitive to unknown CLI args. It was a small number of tests, single digits I think. `test/core/fling/server.cc` is another example that would need some cleanup. This `argc` mutation approach seemed simple, I'm not sure why it's controversial. I don't buy the argument that broad changes should be inherently controversial, I see them in this codebase all the time. This one looks harmless.

If we did a broader cleanup to remove unknown argument sensitivities, we might be able to use absl::Flags instead, which would be another improvement, and another fairly trivial change to this TestEnvironment constructor definition (probably no need for argc and argv). I'd be happy to put that cleanup on a "things to do while compiling" queue. But I don't think that cleanup should block this work.

Reply:
> I think it would be cleaner to fix those tests instead of making this broader change.

I wouldn't label them as things that need fixing. We could definitely change their approaches to arg parsing, but I think that can be done in the other cleanup as you suggested. 

> the change seems to make the API a bit less intuitive

+1 to Craig's comment below, this is a standard approach we can find in our own codebase and dependencies, I did not invent it. I disagree that it's less intuitive.